patches/tModLoader/Terraria/WorldGen.cs.patch

```diff
@@ -170,6 +753,49 @@
  			Main.rand = new UnifiedRandom((int)DateTime.Now.Ticks);
  			WorldFile.LoadWorld(Main.ActiveWorldFileData.IsCloudSave);
  			if (loadFailed || !loadSuccess) {
+@@ -2522,6 +_,15 @@
+ 		public static Task serverLoadWorld() => Task.Factory.StartNew(serverLoadWorldCallBack);
+ 
+ 		public static void clearWorld() {
++			Main.mapTargetX = Math.Max(Main.maxTilesX / 1680, 5);
++			Main.mapTargetY = Math.Max(Main.maxTilesY / 1200, 2);
++			Main.instance.mapTarget = new RenderTarget2D[Main.mapTargetX, Main.mapTargetY];
++			Main.initMap = new bool[Main.mapTargetX, Main.mapTargetY];
++			Main.mapWasContentLost = new bool[Main.mapTargetX, Main.mapTargetY];
++
++			Main.Map = new WorldMap(Main.maxTilesX, Main.maxTilesY);
++			Main.tile = new Tile[Main.maxTilesX, Main.maxTilesY];
++
+ 			Main.ladyBugRainBoost = 0;
+ 			Main.getGoodWorld = false;
+ 			Main.drunkWorld = false;
+@@ -2666,15 +_,17 @@
+ 			noLiquidCheck = false;
+ 			Liquid.numLiquid = 0;
+ 			LiquidBuffer.numLiquidBuffer = 0;
+-			if (Main.netMode == 1 || lastMaxTilesX > Main.maxTilesX || lastMaxTilesY > Main.maxTilesY) {
+-				for (int j = 0; j < lastMaxTilesX; j++) {
+-					float num = (float)j / (float)lastMaxTilesX;
+-					Main.statusText = Lang.gen[46].Value + " " + (int)(num * 100f + 1f) + "%";
+-					for (int k = 0; k < lastMaxTilesY; k++) {
+-						Main.tile[j, k] = null;
+-					}
+-				}
+-			}
++			//Ok, so with this new system this is just not needed anymore. This is pretty radical to get rid of, so please look for bugs related to this closely.
++			//Everything should still be vanilla compatible.
++			//if (Main.netMode == 1 || lastMaxTilesX > Main.maxTilesX || lastMaxTilesY > Main.maxTilesY) {
```

Comment:
Use comment blocks.

Reply:
That shouldn't be a problem, using // is actually preferred since it's "safer"
Here would be a bit more detailed answer https://stackoverflow.com/questions/1745179/c-sharp-coding-style-comments

Reply:
I asked for comment blocks because this is a patch. What you linked has nothing to do with diffs.